### PR TITLE
added gemfile and school controller

### PR DESCRIPTION
**Pull request by harishpathakamuri**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -43,7 +43,12 @@ group :development do
   gem 'spring-watcher-listen', '~> 2.0.0'
 end
 
+gem 'carrierwave', '~> 1.0'
+gem 'carrierwave-video'
+gem 'carrierwave-video-thumbnailer'
 gem 'devise'
+gem "rolify"
+gem 'authority'
```
Comment (madhukosuri) [inline Gemfile]: why do you need authority gem.

File: app/controllers/api/v1/schools_controller.rb [hunk 2]
```diff
@@ -1,96 +1,60 @@
...
-	module V1
-		class SchoolsController < ApplicationController
-			before_action :set_school, only: [:show, :destroy, :update]
-			before_action :authenticate_user!
-
+  module V1
+    class SchoolsController < ApplicationController
+      before_action :set_school, only: [:show, :destroy, :update]
+      before_action :authenticate_user!
    #
```
Comment (madhukosuri) [inline app/controllers/api/v1/schools_controller.rb]: Please remove unwanted character.

File: app/controllers/api/v1/schools_controller.rb [hunk 3]
```diff
@@ -1,96 +1,60 @@
...
-				if School.exists?(params[:id])
-					@school = School.find(params[:id])
-				else
-					return head(:forbidden)
-				end
-			end
-       end
-	end
+       def show
+        puts @current_user.inspect
```
Comment (madhukosuri) [inline app/controllers/api/v1/schools_controller.rb]: Please remove puts here...

File: app/controllers/api/v1/schools_controller.rb [hunk 4]
```diff
@@ -1,96 +1,60 @@
...
+      end
+
+      def index
+        @schools = School.all
+        render json: @schools
+      end
+
+      def create
+        @school = School.new(school_params)
+
```
Comment (madhukosuri) [inline app/controllers/api/v1/schools_controller.rb]: Please remove empty line here, it is unwanted

File: app/controllers/api/v1/schools_controller.rb [hunk 5]
```diff
@@ -1,96 +1,60 @@
...
+        if @school.update_attributes(school_params)
+          render json: "updated"
+        else
+          render json:@school.errors.messages, status: :unprocessable_entity
+        end
+
+      end
+
+      def destroy
+        puts @school.inspect
```
Comment (madhukosuri) [inline app/controllers/api/v1/schools_controller.rb]: Here also please remove puts line.

File: app/controllers/api/v1/schools_controller.rb [hunk 6]
```diff
@@ -1,96 +1,60 @@
...
+        @school.destroy
+        if @school.destroyed?
+          render json: 'destroy successfully'
```
Comment (madhukosuri) [inline app/controllers/api/v1/schools_controller.rb]: Message should school destroyed successfully.

File: app/controllers/api/v1/schools_controller.rb [hunk 7]
```diff
@@ -1,96 +1,60 @@
...
+      def set_school
+        if School.exists?(params[:id])
+          @school = School.find(params[:id])
+        else
+          return head(:forbidden)
+        end
+      end
+    end
+  end
 end
```
Comment (madhukosuri) [inline app/controllers/api/v1/schools_controller.rb:56]: Need one extra line end of the file, because Ubuntu needs

File: Gemfile (hunk 1 above)
Comment (harishpathakamuri) [reply Gemfile]: yes I will remove.

File: app/controllers/api/v1/schools_controller.rb (hunk 7 above)
Comment (harishpathakamuri) [reply app/controllers/api/v1/schools_controller.rb:56]: I think it is fine I will check once in repo.

**Review by madhukosuri — APPROVED** [review]

**Pull request merged**